jena-arq/src/test/java/org/apache/jena/riot/TestJsonLDReader.java

```diff
@@ -18,127 +18,123 @@
 
         // check ds is correct
-        assertJohnDoeIsOK(ds.getDefaultModel());
+        assertJohnDoeIsOK(m);
     }
 
-    /** Not really useful, but one can replace the @context by a URI: in this case, this URI is used when expanding the json
-     * (letting JSON-LD java API taking care of downloading the context. */
-    // well, as of this writing, it doesn't work, as we get a "loading remote context failed"
-    // But it is about the replacing URI, not the replaced one, showing that the mechanism does work
-    @Test public final void overrideAtContextByURITest() throws JsonGenerationException, IOException {
+    /**
+     * Not really useful, but one can replace the @context by a URI: in this case, this URI is used when expanding the json
+     * (letting JSON-LD java API taking care of downloading the context.
+     */
+    @Test
+    public final void overrideJsonLdOptionsAndContextUri() throws JsonGenerationException, IOException {
         // some jsonld using a (fake) pseudo.schema.org's URI as "@context"
         String jsonld = "{\"@id\":\"_:b0\",\"@type\":\"Person\",\"name\":\"John Doe\",\"@context\":\"http://pseudo.schema.org/\"}";
 
         // a subset of schema.org that can be used as @context for jsonld
         String jsonldContext = "\"http://schema.org\"";
 
-        // pass the jsonldContext to the read using a jena Context
-        Context jenaCtx = new Context();
-        Object jsonldContextAsObject = JsonUtils.fromInputStream(new ByteArrayInputStream(jsonldContext.getBytes(StandardCharsets.UTF_8)));
-        jenaCtx.set(RIOT.JSONLD_CONTEXT, jsonldContextAsObject);
-
-        try {
-            // read the jsonld, replacing its "@context"
-            Dataset ds = jsonld2dataset(jsonld, jenaCtx);
-
-            // check ds is correct
-            assertJohnDoeIsOK(ds.getDefaultModel());
-        } catch (RiotException e) {
-            // cf. org.apache.jena.riot.RiotException: loading remote context failed: http://schema.org/
-            // There's a line printed anyway
-            // e.printStackTrace();
-        }
+        JsonLdOptions options = new JsonLdOptions();
+        DocumentLoader dl = new DocumentLoader();
+        dl.addInjectedDoc("http://schema.org", String.format("{%s}", schemaOrgContext()));
+        options.setDocumentLoader(dl);
+
+        // pass the jsonldContext and JsonLdOptions to the read using a jena Context
+        JsonLDReadContext jenaCtx = new JsonLDReadContext();
+        jenaCtx.setJsonLDContext(jsonldContext);
+        jenaCtx.setOptions(options);
+
+        // read the jsonld, replacing its "@context"
+        Model m = jsonld2model(jsonld, jenaCtx);
+
+        // check ds is correct
+        assertJohnDoeIsOK(m);
     }
 
     //
     //
     //
 
-    /**
-     * Reading some jsonld String, using a Context
-     * @return a new Dataset
-     * @throws IOException
-     */
-    private Dataset jsonld2dataset(String jsonld, Context jenaCtx) throws IOException {
```

Comment:
JSON-LD is a dataset format and  can encode datasets (quads) as well as a single graph.

Reply:
In the tests I don't see what the dataset is doing that the model cannot do. But I can keep it if you find it better.